components/tidb_query/src/rpn_expr/impl_miscellaneous.rs

```diff
@@ -0,0 +1,47 @@
+// Copyright 2019 TiKV Project Authors. Licensed under Apache-2.0.
+
+use tidb_query_codegen::rpn_fn;
+
+use crate::codec::data_type::*;
+use crate::Result;
+
+use std::convert::TryFrom;
+use std::net::Ipv4Addr;
+
+#[rpn_fn]
+#[inline]
+pub fn inet_ntoa(arg: &Option<Int>) -> Result<Option<Bytes>> {
+    Ok(arg
+        .and_then(|arg| u32::try_from(arg).ok())
```

Comment:
I see that your implementation is different for the non-vectorized version. Could you change it as well to keep two implementations identical?